tests/util/server/src/npm.rs

```diff
@@ -263,6 +268,15 @@ fn get_npm_package(
       }
     }
 
+    if let Some(deprecated) = version_info.get("deprecated") {
+      log::warn!(
+        "{}:{} is deprecated: {}",
+        serde_json::to_string(&version_info["name"]).unwrap(),
+        serde_json::to_string(&version_info["version"]).unwrap(),
+        serde_json::to_string(deprecated).unwrap()
+      );
+    }
```

Comment:
This is not really what we want to do. Instead of printing a warning in the test server we want to print this info when the package is actually installed.

You should update `sync_resolution_with_fs` function in `cli/npm/managed/resolvers/local.rs` to print this warning. You can follow how it's done for `bin_entries_to_setup`, but that will require to update `deno_npm` crate first to store information about deprecation in `NpmResolutionPackage` struct.

Reply:
Please note that I made two PRs to accomplish the intended behavior.
1) [PR to deno_npm](https://github.com/denoland/deno_npm/pull/61): This updates related structs with the deprecated field.
2) [PR to eszip](https://github.com/denoland/eszip/pull/194): This updates the instantiation of two instances using the modified structs, setting the deprecated field to its default value.

I believe that in order for the checks to be done successfully, first deno_npm PR should be merged then eszip and deno at last.